docs_src/async_sql_databases/sql_app/main.py

```diff
@@ -0,0 +1,60 @@
+from fastapi import Depends, FastAPI, HTTPException
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from . import crud, schemas
+from .database import Base, engine
+
+app = FastAPI()
+
+
+@app.on_event("startup")
+async def start_db():
+    async with engine.begin() as conn:
+        # await conn.run_sync(Base.metadata.drop_all)
+        await conn.run_sync(Base.metadata.create_all)
+
+
+# Dependency
+async def get_db():
+    async with AsyncSession(engine) as session:
+        try:
+            yield session
+        finally:
+            session.close()
```

Comment:
Why do we need to close the sesion if it's inside a context manager?

Reply:
Fixed